### DISCUSSION TITLE
Question about req-rep recv timeout

### DISCUSSION THREAD

**Discussion by mikisch81** [nanomsg/nng#1853] [Q&A] [answered]
Hello,

In req-rep protocol I see that in the client after sending the request while waiting for reply if you don't set recv timeout on the socket and the server disconnects you still wait on `nng_recv`.

Shouldn't it exit with some kind of error like `NNG_ECONNRESET` or something as this is not a server waiting for a message which might arrive sometime in the future, in this case it is a client which already sent a request to the server and is waiting for a response from this specific server, if the the server disconnected for some reason the reply will never arrive.

**Comment by gdamore** [comment, marked as answer]
So the request can be resent later to another server, or the same server might reconnect.  This is part of the self-healing nature of the REQ/REP protocol.  The resend happens underneath the hood.

Obviously if you're using this capability, you should use idempotent requests.

To avoid the retry please try using `NNG_OPT_REQ_RESENDTIME` set to 0.
